api/redhatopenshift/HcpCluster/hcpCluster-models.tsp

```diff
@@ -449,13 +449,17 @@ model NodePoolSpec {
   /** Representation of a autoscaling in a node pool. */
   autoScaling?: NodePoolAutoScaling;
 
+  // This warning means to ensure good customer experience, to not create
+  // poorly defined types. However here the type literarly is the map<string, value>
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-no-record" ""
   /** Labels for the nodes */
```

Comment:
We can probably improve this description a bit

I would change it to:
"K8s labels to propagate to the NodePool Nodes"

being explicit about K8s. The reason for that is to clarify that:
* It refers to K8s labels and not Azure tags
* It refers to the K8s labels associated to the Nodes of the NodePool and not of the NodePool itself
* They are propagated to the Nodes of the NodePool

Reply:
Thank you, this makes it clear.